### PR TITLE
Feature/Add expense

### PR DESCRIPTION
**Pull request by marianbalan**
* Added ExpenseDto entity
* Added ExpenseViewModel entity
* Created POST endpoint for expenses

### CODE REVIEW

File: Service/src/main/java/service/Service.java [hunk 1]
```diff
@@ -33,4 +38,49 @@ public Service(
...
+
+        if (optionalUser.isEmpty()) {
+            throw new ServiceException("The user does not exist.");
+        }
+
+        return new Expense(
+                expenseDto.getAmount(),
+                expenseDto.getCategory(),
+                expenseDto.getDate(),
+                optionalUser.get()
```
Comment (georgianabaietilor) [inline Service/src/main/java/service/Service.java:71]: if you create a constructor in User entity 
  public User(int id){this.id=id;} 
you can change line 71 with 
  new User(expenseDto.getUserId()) 
and you won't need lines 61-65

**Review by georgianabaietilor — APPROVED** [review]

**Review by leonard-pop — COMMENTED** [review]
Do we really need 3 classes for Expense?

File: Service/src/main/java/service/Service.java [hunk 2]
```diff
@@ -33,4 +37,41 @@ public Service(
...
+     * transforms an ExpenseDto object to an Expense object
+     */
+    private Expense expenseDtoToExpense(ExpenseDto expenseDto) throws ServiceException {
+        return new Expense(
+                expenseDto.getAmount(),
+                expenseDto.getCategory(),
+                expenseDto.getDate(),
+                new User(expenseDto.getUserId())
+        );
+    }
```
Comment (oskarpicus) [inline Service/src/main/java/service/Service.java:64]: Maybe this function should be moved to the Expense class in a similar form to the following:
`public static Expense from(ExpenseDto expense) {...}`

File: Rest/src/main/java/controller/ExpenseController.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+public class ExpenseController {
+    @Autowired
+    private IService service;
+
+    @PostMapping("/add-expense")
+    public ResponseEntity<?> create(@RequestBody ExpenseDto expenseDto) {
+        ExpenseViewModel expense;
+
+        try {
+            expense = service.addExpense(expenseDto);
```
Comment (oskarpicus) [inline Rest/src/main/java/controller/ExpenseController.java:24]: The code can be more concise, by directly returning the `service.addExpense(expenseDto);` (unless there is a compelling reason not to)

**Review by oskarpicus — CHANGES_REQUESTED** [review]

File: Service/src/main/java/service/Service.java [hunk 4]
```diff
@@ -33,4 +37,41 @@ public Service(
...
+            throw new ServiceException("An error occurred while saving the expense.");
+        }
+
+        return expenseToExpenseViewModel(expense);
+    }
+
+    /**
+     * transforms an ExpenseDto object to an Expense object
+     */
+    private Expense expenseDtoToExpense(ExpenseDto expenseDto) throws ServiceException {
```
[the remark this answers is not in the document]
Comment (marianbalan) [reply Service/src/main/java/service/Service.java:57]: Forgot to remove it

**Review by oskarpicus — APPROVED** [review]

**Pull request merged**